cmake/Settings.cmake

```diff
@@ -0,0 +1,34 @@
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+# Settings intended to be applied to every Swift target in this package.
+# Analogous to project-level build settings in an Xcode project.
+add_compile_definitions(
+  "$<$<COMPILE_LANGUAGE:CXX>:_SWT_TESTING_LIBRARY_VERSION=unknown (Swift 5.10 toolchain)>"
+  "SWIFT_PACKAGE")
```

Comment:
Not sure `SWIFT_PACKAGE` ought to be set in the CMake context since we might falsely assume it's only set when _actually_ building via SwiftPM. Did you find it to be necessary?

If there are specific places where we currently rely on it in the source, we might want to generalize those guards to not be specific to SwiftPM and this particular define

Reply:
We intentionally don't use it anywhere in swift-testing, so it shouldn't be necessary to define it.

Reply:
Probably correct.  I'll double check this one too.